### PR TITLE
Nc news q11

### PR DESCRIPTION
**Pull request by rubyrubyruby02**

### CODE REVIEW

File: db/sql_playground.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+\c nc_news_test
```
Comment (EddTheDuck) [inline db/sql_playground.sql]: You should probably add this to your .gitignore

File: models/articles.model.js [hunk 2]
```diff
@@ -14,9 +13,13 @@ const fetchArticle = (article_id)=> {
     })
 }
 
-const fetchAllArticles = ()=> {
+const fetchAllArticles = (topic)=> {
+
+    console.log(topic, "in model")
```
Comment (EddTheDuck) [inline models/articles.model.js]: remove console.log

File: models/articles.model.js [hunk 3]
```diff
@@ -27,20 +30,29 @@ const fetchAllArticles = ()=> {
     a.article_img_url
     FROM articles a 
     INNER JOIN comments c 
-    ON a.article_id = c.article_id
-    GROUP BY a.article_id
-    ORDER BY a.created_at DESC;`)
+    ON a.article_id = c.article_id `
+
+    if(topic){
+        console.log("in if block")
```
Comment (EddTheDuck) [inline models/articles.model.js]: remove console.log

File: models/articles.model.js [hunk 4]
```diff
@@ -27,20 +30,29 @@ const fetchAllArticles = ()=> {
...
+        baseQueryString+=`WHERE a.topic = $1 `
+        paramsArrayNoSQLInj.push(topic)
+    }
+
+    baseQueryString+=`GROUP BY a.article_id
+    ORDER BY a.created_at DESC`
+
+    console.log(baseQueryString)
```
Comment (EddTheDuck) [inline models/articles.model.js]: remove console.log

File: models/articles.model.js [hunk 5]
```diff
@@ -27,20 +30,29 @@ const fetchAllArticles = ()=> {
...
+
+    return db.query(baseQueryString, paramsArrayNoSQLInj)
     .then((result)=> {
+        console.log(result.rows)
```
Comment (EddTheDuck) [inline models/articles.model.js]: remove console.log

File: output.txt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+You are now connected to database "nc_news_test" as user "ruby".
```
Comment (EddTheDuck) [inline output.txt]: You should add this file to your .gitignore

File: __tests__/integration.tests.js [hunk 7]
```diff
@@ -560,12 +560,30 @@ describe('Q11 QUERIES GET /api/articles', ()=> {
...
+    })
+    test('Status 200 - can handle a topic, sort_by and order', ()=> {
+        return request(app)
+        .get('/api/articles?topic=mitch&sort_by=article_id&order=asc')
+        .expect(200)
+        .then((response)=> {
+
+            expect(response.body.articles).toHaveLength(12)
+            expect(response.body.articles).toBeSortedBy('article_id', {ascending: true})
+        })
```
Comment (EddTheDuck) [inline __tests__/integration.tests.js:582]: If we are querying topic = mitch why don't we assert that that's what we get back ?

File: __tests__/integration.tests.js [hunk 8]
```diff
@@ -568,22 +568,50 @@ describe('Q11 QUERIES GET /api/articles Ascending & Descending', ()=> {
...
+    test('Status 400 Bad request if not asc or desc and another string is input', ()=> {
+        return request(app)
+        .get('/api/articles?order=notAscOrDesc')
+        .expect(400)
+        .then((response)=> {
+            expect(response.body.msg).toBe("Bad request")
+        })
+    })
+    test('Status 400 Bad request - handles multiple wrong inputs', ()=> {
+        return request(app)
```
Comment (EddTheDuck) [inline __tests__/integration.tests.js]: This test should be split up to be more thorough

Comment (EddTheDuck) [reply __tests__/integration.tests.js]: For example, 
Status 400. invalid sort_by query, e.g. ?sort_by=bananas
Status 400. invalid order query, e.g. ?order=bananas

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
I think just refactor your tests to be more thorough please, but very nearly there and overall great work :)

File: __tests__/integration.tests.js (hunk 8 above)
Comment (rubyrubyruby02) [reply __tests__/integration.tests.js]: Both of these tests are separated above so I think it means this final test is probably not needed. I was aiming to test multiple wrong inputs all together but maybe this isn't needed as one wrong input will throw the error.

**Pull request merged**